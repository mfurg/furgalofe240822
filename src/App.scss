.App {
  display: flex;
  justify-content: center;
  align-items: center;
}

button {
  max-width: 100px;
  margin: 30px auto;
}

dl {
  display: flex;
  background-color: white;
  flex-direction: column;
  width: 100%;
  max-width: 700px;
  position: relative;
  padding: 20px;
}

dt {
  align-self: flex-start;
  align-content: center;
  width: 100%;
  font-weight: 700;
  display: block;
  text-align: center;
  font-size: 20px;
  font-weight: 700;
  margin-bottom: 20px;
  margin-left: 75px;
}

.text {
  font-weight: 600;
  display: flex;
  align-items: center;
  height: 40px;
  width: 150px;
  background-color: white;
  position: absolute;
  left: 0;
  justify-content: flex-end;
}

.percentage {
  font-size: .8em;
  line-height: 1;
  text-transform: uppercase;
  width: 100%;
  height: 40px;
  margin-left: 150px;
  background: repeating-linear-gradient(
  to right,
  #ddd,
  #ddd 1px,
  #fff 1px,
  #fff 5%
);
  
  &:after {
    content: "";
    display: block;
    background-color: #3d9970;
    width: 50px;
    margin-bottom: 10px;
    height: 90%;
    position: relative;
    top: 50%;
    transform: translateY(-50%);
    transition: background-color .3s ease;
    cursor: pointer;
  }
  &:hover,
  &:focus {
    &:after {
       background-color: #aaa; 
    }
  }
}

@for $i from 1 through 100 {
  .percentage-#{$i} {
    &:after {
      $value: ($i * 1%);
      width: $value;
    }
  }
}

@for $i from 1 through 100 {
  .margin-#{$i} {
    &:after {
      $value: ($i * 1%);
      margin-left: $value;
    }
  }
}

html, body {
  height: 500px;
  font-family: 'Roboto', sans-serif;
  font-size: 16px;
  color: #333;
}