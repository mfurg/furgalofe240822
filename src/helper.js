export const percentCalc = (time, total) => {
    return Math.round(time/(total/100))
  }