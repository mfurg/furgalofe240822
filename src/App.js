import { useState, useEffect } from 'react';
import './App.scss';
import { Button } from './Button';
import { percentCalc } from './helper';

function App() {
  const [data, setData] = useState([{ name: "Landing Page", time: 7.4 },
                { name: "Configurator", time: 0.2 },
                { name: "Check-out", time: 7.0 },
                { name: "Deal", time: 3.8 }])
  let totalMargin = 0;
  const totalTime = data.reduce((accumulator, object) => {
    return accumulator + object.time;
  }, 0).toFixed(2);

  data.forEach((item) => {
    item.margin = totalMargin;
    totalMargin += percentCalc(item.time, totalTime)
  });

  useEffect(()=>{
    const timer = setInterval(setRandomData, (60000*0.53));
  
    return () => {
      clearInterval(timer);
    };
  },[]);
  

  const setRandomData = () => {
    let newData = data.map((item) => {
      return {...item, time: Math.ceil(Math.random()*30)};
   })
   setData(newData)
  }
 
  return (
    <div className="App">
      <dl>
        <dt>
          SPEND TIME ( {totalTime} SECONDS)
        </dt>
        {data.map((item, i) => 
          <dd key={i} className={`percentage margin-${item.margin} percentage-${percentCalc(item.time, totalTime)}`}>
            <span className="text">
              {item.name} {item.time}s
            </span>
          </dd>)}
          <dt>
            <Button setRandomData={setRandomData}>Random</Button>
          </dt>
      </dl>
    </div>
  );
}

export default App;
