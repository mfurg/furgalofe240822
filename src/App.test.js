import { cleanup, render } from "@testing-library/react";
import App from "./App";
import { percentCalc } from "./helper";

afterEach(cleanup);

describe("Calculator tests", () => {
  test('percent calculate', () => {
    expect(percentCalc(7,100)).toBe(7);
  });
 })

it("renders", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

