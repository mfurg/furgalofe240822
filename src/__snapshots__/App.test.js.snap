// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <dl>
      <dt>
        SPEND TIME ( 18.40 SECONDS)
      </dt>
      <dd
        class="percentage margin-0 percentage-40"
      >
        <span
          class="text"
        >
          Landing Page 7.4s
        </span>
      </dd>
      <dd
        class="percentage margin-40 percentage-1"
      >
        <span
          class="text"
        >
          Configurator 0.2s
        </span>
      </dd>
      <dd
        class="percentage margin-41 percentage-38"
      >
        <span
          class="text"
        >
          Check-out 7s
        </span>
      </dd>
      <dd
        class="percentage margin-79 percentage-21"
      >
        <span
          class="text"
        >
          Deal 3.8s
        </span>
      </dd>
      <dt>
        <button>
          Random
        </button>
      </dt>
    </dl>
  </div>
</DocumentFragment>
`;
